chalice/chalicelib/matrix_handler.py

```diff
@@ -0,0 +1,165 @@
+import os
+import tempfile
+import time
+import traceback
+import loompy
+
+from abc import ABC, abstractmethod
+from subprocess import call
+from typing import List
+from chalicelib import get_mtx_paths
+from chalicelib.config import MERGED_MTX_BUCKET_NAME, TEMP_DIR, hca_client, \
+    s3_blob_store, logger
+from chalicelib.request_handler import RequestHandler, RequestStatus
+
+
+class MatrixHandler(ABC):
+    """
+    A generic matrix handler for matrices concatenation
+    """
+
+    def __init__(self, suffix) -> None:
+        self._suffix = suffix
+
+    def _download_mtx(self, bundle_uuids: List[str], temp_dir: str) -> List[str]:
+        """
+        Filter for the matrix files within bundles, and download them locally.
+        :param bundle_uuids: A list of bundle uuids.
+        :param temp_dir: A temporary directory for storing all downloaded matrix files.
+        :return: A list of downloaded matrix files paths.
+        """
+        # Filter and download only matrices file that satisfies a specific suffix within bundles
+        for bundle_uuid in bundle_uuids:
+            dest_name = os.path.join(temp_dir, bundle_uuid)
+
+            try:
+                logger.info("Downloading matrices from bundle, {}, into {}."
+                            .format(bundle_uuid, dest_name))
+
+                hca_client.download(
+                    bundle_uuid=bundle_uuid,
+                    replica="aws",
+                    dest_name=dest_name,
+                    metadata_files=(),
+                    data_files=("*{}".format(self._suffix),)
+                )
+            except Exception as e:
+                raise e
+
+        # Get all downloaded mtx paths from temp_dir
+        local_mtx_paths = get_mtx_paths(temp_dir, self._suffix)
+
+        logger.info("Done downloading %d matrix files.", len(local_mtx_paths))
+
+        return local_mtx_paths
+
+    @abstractmethod
+    def _concat_mtx(self, mtx_paths: List[str], out_file: str) -> None:
+        """
+        Concatenate a list of matrices, and save into a new file on disk.
+        :param mtx_paths: A list of downloaded local matrix files paths.
+        :param out_file: Path to the concatenated matrix.
+        """
+
+    def _upload_mtx(self, path: str, request_id: str):
+        """
+        Upload a matrix file into an s3 bucket.
+        :param path: Path of the merged matrix.
+        :param request_id: Merge request id.
+        """
+        logger.info("%s", "Uploading \"{}\" to s3 bucket: \"{}\".".format(
+            os.path.basename(path),
+            MERGED_MTX_BUCKET_NAME
+        ))
+
+        with open(path, "rb") as merged_matrix:
+            try:
+                s3_blob_store.upload_file_handle(
+                    bucket=MERGED_MTX_BUCKET_NAME,
+                    key=request_id + self._suffix,
+                    src_file_handle=merged_matrix
+                )
+            except Exception as e:
```

Comment:
The
```python
try:
    ...
except Exception as e:
    raise e
```
pattern appears a number of times, but it doesn't really do anything does it? I would omit `try`/`except` block for these. The behavior is the same, and it won't look like there's special exception handling happening.

Reply:
Okay. I will modify it to make the code cleaner.